pkg/controller/apimanagerbackup/apimanagerbackup_logic_reconciler.go

```diff
@@ -0,0 +1,398 @@
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *APIManagerBackupLogicReconciler) reconcileBackupSystemFileStoragePVCToPVCJob() (reconcile.Result, error) {
+	desired := r.apiManagerBackup.BackupSystemFileStoragePVCToPVCJob()
+	if desired == nil {
+		return reconcile.Result{}, nil
+	}
+
+	// TODO create Mutator function for Jobs??
+	existing := &batchv1.Job{}
+	if r.cr.SystemFileStorageBackupStepFinished() {
+		return reconcile.Result{}, nil
+	}
+
+	// Check if backup substep has completed
+	if !r.cr.SystemFileStorageBackupSubStepFinished() {
+		err := r.ReconcileResource(existing, desired, reconcilers.CreateOnlyMutator)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		backupFinished := true
+		if existing.Status.Succeeded != *desired.Spec.Completions {
+			r.Logger().Info("Job has still not finished", "Job Name", desired.Name, "Actively running Pods", existing.Status.Active, "Failed pods", existing.Status.Failed)
+			return reconcile.Result{Requeue: true}, nil
+		}
+
+		r.cr.Status.SystemFileStorageBackupSubStepFinished = &backupFinished
+		err = r.UpdateResourceStatus(r.cr)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		r.Logger().Info("Job finished successfully. Requeing", "Job Name", desired.Name)
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	// Check if cleanup substep has completed
+	if !r.cr.SystemFileStorageCleanupSubStepFinished() {
+		// TODO delete performed with ReconcileResource does not have the option to delete
+		// dependants which means that if the Job is deleted the Pod is left there
+		// Think of a way to do this. Maybe instead of using ReconcileResource perform
+		// the manual deletion.
+		// Apparently leaving the pod in terminated is not a problem regarding the volume
+		// It seems that the volume can be specified in several Pods if only one of them
+		// is really using (this is, the others are terminated)
+		// In any case we should fix it. I've observed for example that you
+		// cannot delete the pvc manually if it's being used by some Pod, even if
+		// it is terminated
+		common.TagToObjectDeleteWithPropagationPolicy(desired, metav1.DeletePropagationForeground)
+
+		err := r.ReconcileResource(existing, desired, reconcilers.CreateOnlyMutator)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		err = r.GetResource(common.ObjectKey(desired), existing)
+		if err != nil && !errors.IsNotFound(err) {
+			return reconcile.Result{}, err
+		}
+
+		backupFinished := true
+		r.cr.Status.SystemFileStorageCleanupSubStepFinished = &backupFinished
+		err = r.UpdateResourceStatus(r.cr)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		r.Logger().Info("Job cleant up successfully. Requeuing", "Job Name", desired.Name)
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *APIManagerBackupLogicReconciler) reconcileBackupCompletion() (reconcile.Result, error) {
+	if !r.cr.BackupCompleted() {
```

Comment:
I agree. It somehow seems more robust checking that so it does not try to perform the update always just in case there was some error before.
If you look at the TODO I left there, some of my comments where:
```
// TODO make this more robust only setting it in case all substeps have been completed?
// It might be a little bit redundant because the steps are checked during the reconciliation
```
So I was/am not sure whether it would be a good idea that the state is consistent at that point. It _should_ be at that point but it might not in reality (if there is some error in our side). It's kind of, is it worth it adding this sort of redundancy for added security?

What do you think? I could see both solutions and I have points in favour and against each of them